src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,155 @@
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    public static final int MIN_USER_AGE = 18;
+    public static final int MIN_PASSWORD_LENGTH = 6;
+    private RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+    private StorageDaoImpl storageDao = new StorageDaoImpl();
+    private User user;
+
+    @Test
+    void register_sameLogin_NotOk() {
+        User testUser = new User();
+        testUser.setAge(20);
+        testUser.setLogin("Mersedes");
+        testUser.setPassword("king_1975");
+        storageDao.add(testUser);
+        User user = new User();
+        user.setLogin(testUser.getLogin());
+        user.setPassword("somepassword");
+        user.setAge(25);
+        assertThrows(RuntimeException.class, () ->
+                registrationService.register(user).equals(storageDao.get(user.getLogin())));
+    }
+
+    @Test
+    void register_nullPassword_NotOk() {
+        user = new User();
+        user.setAge(19);
+        user.setLogin("kijuljhbhzt");
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_loginIsNull_NotOk() {
+        User user = new User();
+        user.setPassword("somevalidpassword");
+        user.setAge(19);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_ageIsNull_NotOk() {
+        user = new User();
+        user.setLogin("kijäloonjiuhzt");
+        user.setPassword("somevalid");
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_userIsNull_NotOk() {
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_negativeAge_NotOk() {
+        user = new User();
+        user.setAge(-1);
+        user.setLogin("kijupoohzt");
+        user.setPassword("newidiaa");
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_zeroAge_NotOk() {
+        user = new User();
+        user.setAge(0);
+        user.setLogin("kijüpijuhzt");
+        user.setPassword("jdhszdgtr");
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_ageLessThatMinAge_NotOk() {
+        user = new User();
+        user.setAge(17);
+        user.setLogin("kijuhzt");
+        user.setPassword("jdhszdgtr");
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_minAge_Ok() {
+        user = new User();
+        user.setAge(MIN_USER_AGE);
+        user.setLogin("kjgvfujzfuzfpp");
+        user.setPassword("ljhzvgljz");
+        assertNotNull(registrationService.register(user));/////
```

Comment:
```suggestion
        assertNotNull(registrationService.register(user));
```

lets remove this comments, and we also can add check for another fields: if age, login and password in return user the same as in incoming